Re write routing to query based system

Hyperkid123 opened the pull request:
### Changes
- no visual changes
- routing is switched to query based system
  - instead of insane nesting and params ambiguity, every entity id is clearly defined by query param in URL
- created new consistent breadcrumbs system
  - breadcrumbs can be generated from URL pathname
  - every root page controls when to update it

rvsia commented on src/smart-components/portfolio/portfolio-item-detail/detail-toolbar-actions.js:
```diff
@@ -20,25 +20,18 @@ const DetailToolbarActions = ({
   setOpen,
   isFetching
 }) => {
```
unnecessary block

rvsia commented:
@Hyperkid123 Breadcrumbs do not work properly

![badbreadcrumbs](https://user-images.githubusercontent.com/32869456/73356908-a98a5200-429b-11ea-9279-e6cb230f79e3.gif)

Hyperkid123 commented on src/helpers/shared/breadcrumbs-creators.js:
```diff
@@ -0,0 +1,47 @@
+    getState().portfolioReducer.portfolioItem.portfolioItem.name,
+  '/portfolio/portfolio-item/edit-survey': () => 'Edit survey',
+  '/portfolio/portfolio-item/add-products': () => 'Add products',
+  '/platform': (getState) => getState().platformReducer.selectedPlatform.name,
```
Nope

Hyperkid123 commented on src/presentational-components/platform/platform-item.js:
```diff
@@ -5,17 +5,12 @@ import { Card, CardHeader, CardFooter } from '@patternfly/react-core';
-        style={{ height: 40 }}
-        sourceId={props.source_id}
-      />
+      <CardIcon src={src} style={{ height: 40 }} />
```
In future

Hyperkid123 commented on src/presentational-components/shared/top-toolbar.js:
```diff
@@ -9,7 +9,7 @@ import {
 import clsx from 'clsx';
 import { ToolbarTitlePlaceholder } from './loader-placeholders';
-import CatalogBreadcrumbs from './breadcrubms';
+import CatalogBreadcrumbs from '../../smart-components/common/catalog-bread-crumbs';
```
lol

Hyperkid123 commented:
@rvsia there was a missing default trigger for breadcrumbs re-render. Should be fixed soon.

Hyperkid123 commented on src/router.js:
```diff
@@ -0,0 +1,24 @@
+const Router = () => {
+  return (
+    <AppContext.Provider value={{ release }}>
+      <BrowserRouter basename={`${release}${pathName[0]}/${pathName[1]}`}>
```
I need the release and I cant do that with the function from front-end components

rvsia commented:
@Hyperkid123 Tabs breadcrumbs do not work properly

![breadcrumbs2](https://user-images.githubusercontent.com/32869456/73361364-1d7d2800-42a5-11ea-91d5-6fe87c9a370b.gif)

Hyperkid123 commented:
@rvsia I think we should remove the breadcrumbs from tabs since it is not a "full" route

rvsia commented:
@Hyperkid123 Yeah, I totally agree. :+1:

Hyperkid123 commented:
@rvsia ok so I moved the strings to constants and removed the breadcrumbs from the tabs. Also fixed some code smells. It should be OK now but I would appreciate if you could click through the app again.

rvsia commented:
@Hyperkid123 

Yes, I will check it tomorrow. And don’t worry about the size of this PR, last time I was working on breadcrumbs, it took only 8 months.. 😸

lgalis commented:
@Hyperkid  - Copy portfolio also fails

lgalis commented:
@Hyperkid123 - copy portfolio also fails

rvsia commented on src/smart-components/common/catalog-breadcrumbs.js:
```diff
@@ -0,0 +1,34 @@
+          nav
+          exact
+          key={pathname}
+          className="pf-c-breadcrumb__item"
```
You should check here if it is active, otherwise you dont want to add `cursor: pointer` to it

Hyperkid123 replied:
Fixed that with CSS.

Hyperkid123 commented:
@lgalis I fixed the issue with routing. @rvsia styling was fixed.

Hyperkid123 commented:
Share/unshare notifications are missing

rvsia commented:
@Hyperkid123 
![addproducts](https://user-images.githubusercontent.com/32869456/73456111-977ae300-4371-11ea-9b0a-36035daaf3ab.gif)

Hyperkid123 commented:
@rvsia fixed
